### PR TITLE
Add gcloud role and related tasks

### PR DESCRIPTION
**Pull request by santhosh-tg**
Adding gcloud related role and updating CSP related  gcloud tasks.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/gcp-cloud-storage/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+local_file_or_folder_path: ""
+
+# The name of the file in gcloud storage after uploading from local path
+# The name of the file in gcloud storage that has to be downloaded
+# Example -
+# dest_file_name: "/myfile-blob.json"
+dest_file_name: ""
+
+
+# The folder path in gcloud storage to upload the files starting from the root of the container
```
Comment (keshavprasadms) [inline ansible/roles/gcp-cloud-storage/defaults/main.yml:37]: This line still has reference as container. Please change it to bucket.

File: ansible/roles/gcp-cloud-storage/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+dependencies:
```
Comment (keshavprasadms) [inline ansible/roles/gcp-cloud-storage/meta/main.yml:1]: Let's remove the meta directory. The expectation is that a user runs the bootstrap job and installs the required packages first.

File: ansible/roles/gcp-cloud-storage/tasks/gcloud-auth.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+- name: Copy service account key file
+  copy: 
+    content: "{{ gcp_storage_key_file }}" 
+    dest: /tmp/gcp_storage_key_file.json
```
Comment (keshavprasadms) [inline ansible/roles/gcp-cloud-storage/tasks/gcloud-auth.yml:5]: Can we prefix this with some unique value that is specific for this run only only? For instance, if two jobs run at the same time on the same VM, chances are they will be overwritten. Maybe an epoch timestamp can give us certain level uniqueness.

**Review by keshavprasadms — CHANGES_REQUESTED** [review]
Please also add the private repo template variables required for this change.

**Comment by santhosh-tg** [comment]
@keshavprasadms  addressed the comments, please review

**Review by keshavprasadms — APPROVED** [review]
Looks good. In the next PR, please also include the code to install gcloud packages using the bootstrap Jenkins job.

**Pull request merged**